### PR TITLE
feat(js/plugins/vertexai): strict json mode

### PR DESCRIPTION
**Pull request by cabljac**
This PR adds strict json mode support to Vertex AI plugin in `next`

Checklist (if applicable):
- [ ] Tested (manually, unit tested, etc.)
- [ ] Changelog updated
- [ ] Docs updated

### CODE REVIEW

File: js/plugins/vertexai/src/gemini.ts [hunk 1]
```diff
@@ -398,7 +398,10 @@ function fromGeminiFunctionResponsePart(part: GeminiPart): Part {
 }
 
 // Converts vertex part to genkit part
-function fromGeminiPart(part: GeminiPart): Part {
+function fromGeminiPart(part: GeminiPart, jsonMode: boolean): Part {
+  if (jsonMode && part.text !== undefined) {
+    return { data: JSON.parse(part.text) };
```
Comment (pavelgj) [inline js/plugins/vertexai/src/gemini.ts:403]: is `JSON.parse` safe here? Do we need to guard agains markdown and extraneous text? Or does the gemini json mode guarantee to ouput clean json only?

File: js/plugins/vertexai/src/gemini.ts [hunk 2]
```diff
@@ -517,11 +521,18 @@ export function geminiModel(
         }
       }
 
+      const tools = request.tools?.length
+        ? [{ functionDeclarations: request.tools?.map(toGeminiTool) }]
+        : [];
+
+      // Cannot use tools and function calling at the same time
+      const jsonMode =
+        (request.output?.format === 'json' || !!request.output?.schema) &&
```
Comment (pavelgj) [inline js/plugins/vertexai/src/gemini.ts:530]: this implementation does not support passing output schema as model responseSchema?
https://ai.google.dev/gemini-api/docs/structured-output?lang=node#supply-schema-in-config

File: js/plugins/vertexai/src/gemini.ts (hunk 1 above)
Comment (cabljac) [reply js/plugins/vertexai/src/gemini.ts:403]: it's not safe, we need to use lenient parsing, there's an open issue for it. Working on a fix (this is also in main)

File: js/plugins/vertexai/src/gemini.ts (hunk 2 above)
Comment (cabljac) [reply js/plugins/vertexai/src/gemini.ts:530]: Yeah we need to add this feature in too, do you want these changes on this PR? This PR was in theory just to sync with what went into `main`

Comment (cabljac) [reply js/plugins/vertexai/src/gemini.ts:530]: i'll add as commits here actually, that makes sense

File: js/plugins/vertexai/src/gemini.ts (hunk 1 above)
Comment (cabljac) [reply js/plugins/vertexai/src/gemini.ts]: I think it will guarantee clean json but not complete JSON, but will double check

**Comment by cabljac** [comment]
closing as resolved in https://github.com/firebase/genkit/pull/1149/files

responseSchema will be dealt with in a new PR

**Pull request closed without merging**